Solutions/Problem_Statement_81_Solution.java

```diff
@@ -0,0 +1,150 @@
+/*
+ QUESTION :
+ 
+ Suppose there are n points along a straight trail, while a runner runs sprints of intervals between those points.
+The training plan is an array a[], which implies the runner should run from point a[i] to point a[i+1]. For example, given n = 10, a = [2, 4, 1, 2]. The runner should run from point 2 to point 4, then turn back from point 4 to point 1 (turning back form one point will count in frequency), and then from point 1 to point 2. Find the point that visited the most by the runner after he finished training, i.e., in the above example, point 2 is the most visited. If more than one point is visited the most, find the point with minimum index.
+constraints:
+
+1 <= n <= 100,000
+1 <= a[i] <=n
+size of array a <=100,000
+
+Input:
+the input is of 2 lines,
+the first line denotes points on the line (n) followed by number of elements in array,
+and second-line represents the array elements.
+
+output:
+Single integer, which denotes the smallest point having a maximum frequency.
+
+sample input 1
+10 4
+2 4 1 2
+
+sample output 1
+2
+
+sample input 2
+20 15
+1 10 12 5 15 10 2 5 7 9 11 6 9 10 7
+
+sample output 2
+7
+
+Expected Time Complexity O(n)
+Expected Space Complexity O(n)
+
+Explanation of sample input 1
+so the point on line are 10,
+number of elements in array is 4, and the array is = {2, 4, 1, 2}
+ * 
+ */
+
+package mozilla;
+
+import java.util.*;
+
+public class Problem_Statement_81_Solution {
+
+	public static void main(String[] args) {
+		
+		Scanner sc = new Scanner(System.in);
+		
+		//Accepting input data from user
+		System.out.println("Enter n : ");
```

Comment:
please try not to print anything in the code other than output. For example, when taking input you are printing "Enter n: " than taking input, rather than doing this just take input from the console.